### PR TITLE
add default conversion handling for imagestream kind

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton ptal.
@mfojtik fyi.

**Comment by bparees** [comment]
[test]

File: pkg/build/api/v1beta1/conversion.go [hunk 1]
```diff
@@ -197,6 +221,18 @@ func init() {
 					out.Tag = tag
 				case "DockerImage":
 					out.Image = in.From.Name
+				case "ImageStream":
+					out.From = &kapi.ObjectReference{
+						Name:      in.From.Name,
+						Namespace: in.From.Namespace,
+						Kind:      in.From.Kind,
+					}
+				default:
```
Comment (smarterclayton) [inline pkg/build/api/v1beta1/conversion.go]: Sending any other kind is a validation error, not a conversion overwrite.  We should be rejecting unknown kinds in the core validation I think.

Comment (bparees) [inline pkg/build/api/v1beta1/conversion.go]: hm.  ok, i'll change it to "if empty, do this"

**Comment by bparees** [comment]
@smarterclayton updated. better?

**Comment by smarterclayton** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**